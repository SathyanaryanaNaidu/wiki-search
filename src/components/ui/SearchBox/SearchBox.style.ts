import css from "styled-jsx/css";

const searchBoxStyles = css`
  .search-box {
    display: flex;
    justify-content: center;
    align-items: center;
    width: 100%;
    max-width: 600px;
    margin: 20px auto;
    position: relative;

    input {
      display: flex;
      width: 100%;
      font-size: 24px;
      text-align: center;
      padding: 16px;
      outline: none;
      border: 2px solid black;
      border-radius: 8px;
    }

    button {
      outline: none;
      border: none;
      position: absolute;
      right: 9px;
      height: 100%;
      background: transparent;
      text-decoration: underline;
      cursor: pointer;
    }
  }
`;

export default searchBoxStyles;
