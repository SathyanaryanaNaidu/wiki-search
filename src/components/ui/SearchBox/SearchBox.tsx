import React from "react";
import searchBoxStyles from "./SearchBox.style";
import {
  SEARCH_BOX_CLEAR_BUTTON,
  SEARCH_BOX_PLACEHOLDER,
} from "@/src/constants/labels";

type SearchBoxProps = {
  value: string;
  handleInputChange: (a: string) => void;
  handleClearInput: () => void;
};

const SearchBox = ({
  value,
  handleInputChange,
  handleClearInput,
}: SearchBoxProps) => {
  return (
    <div className="search-box">
      <input
        value={value}
        onChange={handleInputChange}
        placeholder={SEARCH_BOX_PLACEHOLDER}
      />
      {value && value.length > 0 && (
        <button onClick={handleClearInput}> {SEARCH_BOX_CLEAR_BUTTON} </button>
      )}
      <style jsx>{searchBoxStyles}</style>
    </div>
  );
};

export default SearchBox;
