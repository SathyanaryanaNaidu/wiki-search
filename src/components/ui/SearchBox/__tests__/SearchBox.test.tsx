import { render } from "@testing-library/react";
import SearchBox from "../SearchBox";

it("renders searchBox unchanged", () => {
  const { container } = render(
    <SearchBox
      value=""
      handleInputChange={jest.fn()}
      handleClearInput={jest.fn()}
    />
  );
  expect(container).toMatchSnapshot();
});
