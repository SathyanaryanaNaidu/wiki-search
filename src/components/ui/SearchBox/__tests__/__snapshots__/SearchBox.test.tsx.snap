// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders searchBox unchanged 1`] = `
<div>
  <div
    class="jsx-afdd1919b9d522ba search-box"
  >
    <input
      class="jsx-afdd1919b9d522ba"
      data-testid="search-input"
      placeholder="search wikipedia here"
      value=""
    />
  </div>
</div>
`;
